control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -4761,6 +4842,29 @@ func TestGetTokenMetaFromDescription(t *testing.T) {
 	}
 }
 
+func TestMapAddresses(t *testing.T) {
+	addresses := corev1.EndpointSubset{
+		Addresses: []corev1.EndpointAddress{
+			{Hostname: "host1"},
+			{Hostname: "host2"},
+		},
+		NotReadyAddresses: []corev1.EndpointAddress{
+			{Hostname: "host3"},
+			{Hostname: "host4"},
+		},
+	}
```

Comment:
You can add a few more cases here where all addresses are ready or all addresses are not ready and compare them against the output.